posts/formalizing-isocrystals-in-lean.md

```diff
@@ -0,0 +1,211 @@
+
+
+# Frobenius-semilinear automorphisms of the fraction field of the Witt vectors
+
+
+So, for the fraction field $K$ of the Witt vectors of an algebraically closed field $k$, what is this multiplicative subgroup
+$ \\{ \varphi(b) / b : b \in K^\* \\}$
+of the nonzero elements 
+$K^\*$ ?
+Also, the cosets of this subgroup in $K^*$ are precisely in bijection with the isocrystals of dimension 1, so what is a complete set of representatives of the cosets?
+
+At this point, we found a [2011 MathOverflow post](https://mathoverflow.net/questions/62468/about-frobenius-of-witt-vectors) from a drive-by pseudonym asking precisely this reformulated question.  ["Asker,"](https://mathoverflow.net/users/14572/asker) you gave no motivation for this rather specific and technical question ... we're still wondering whether you came up with it by the same route that we did!
+
+So here is the answer.  The Witt vectors $\mathbb{W}(k)$ live inside their fraction field $K$ and there is a distinguished multiplicative subgroup $\mathbb{W}(k)^\times$ of $K^*$, the original units of $\mathbb{W}(k)$.  In the representation of $\mathbb{W}(k)$ as a sequence of elements of $k$, the units are [precisely](https://github.com/leanprover-community/mathlib/blob/lean-3.44.1/src/ring_theory/witt_vector/discrete_valuation_ring.lean#L58) the sequences whose first element is nonzero.
+
+The $p$-typical Witt vector $(0,1,0,\ldots)$ is referred to in $\mathbb{W}(k)$ as $p$ (it actually ends up being the sum of $p$ copies of $1=(1,0,0,\ldots)$, under the crazy Witt vector addition).  Multiplication (in the crazy Witt vector sense) by $p$ sends any Witt vector to a Witt vector with $0$ in the first position, and more generally multiplication by $p^m$ for any $m\in \mathbb{N}$ sends any Witt vector to a Witt vector with $m$ leading zeros. <!-- double-check how it works --> In fact, every nonzero Witt vector is [of the form $p^m w$](https://github.com/leanprover-community/mathlib/blob/lean-3.44.1/src/ring_theory/witt_vector/discrete_valuation_ring.lean#L117) for some $m\in\mathbb{N}$ and some $w\in\mathbb{W}(k)^\times$.  The statement for the fraction field $K$ is only slightly more complicated -- every nonzero element of $K$ is of the form $p^m w$ for some $m\in\mathbb{Z}$ and some Witt vector $w\in\mathbb{W}(k)^\times$. 
```

Comment:
I personally always picture Witt vectors as trailing off to the left. Just like with `p`-adic "decimal expansions" of `p`-adic integers.

Reply:
Note that
```
/-- The composition of Frobenius and Verschiebung is multiplication by `p`. -/
lemma frobenius_verschiebung (x : 𝕎 R) :
  frobenius (verschiebung x) = x * p :=
by { ghost_calc x, ghost_simp [mul_comm] }
```
tells you even more precisely what multiplication by `p` looks like, when `R` has characteristic `p`: you shift all the coefficients, introducing a `0` in the first position, and then you raise every coefficient to the power `p`. (Of course `0^p = 0`, so the first coeff is still `0`.)

Reply:
Why do you use $^\ast$ to denote units of the field, and $^\times$ for units in the ring?

Reply:
Implemented the two notational suggestions, thanks!  Regarding @jcommelin's comment about explaining what multiplication by `p` looks like, this seems like maybe too much detail for the blog post, but we added a link to that theorem on that sentence.  OK?